web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,196 @@
+# PDEP-13: The pandas Logical Type System
+
+- Created: 27 Apr 2024
+- Status: Under discussion
+- Discussion: [#58141](https://github.com/pandas-dev/pandas/issues/58141)
+- Author: [Will Ayd](https://github.com/willayd),
+- Revision: 3
+
+## Abstract
+
+This PDEP proposes a logical type system for pandas which will decouple user semantics (i.e. _this column should be an integer_) from the pandas implementation/internal (i.e. _we will use NumPy/pyarrow/X to store this array_).By decoupling these through a logical type system, the expectation is that this PDEP will:
+
+  * Abstract underlying library differences from end users
+  * More clearly define the types of data pandas supports
+  * Allow pandas developers more flexibility to manage type implementations
+  * Pave the way for continued adoption of Arrow in the pandas code base
+
+## Background
+
+When pandas was originally built, the data types that it exposed were a subset of the NumPy type system. Starting back in version 0.23.0, pandas introduced Extension types, which it also began to use internally as a way of creating arrays instead of exclusively relying upon NumPy. Over the course of the 1.x releases, pandas began using pyarrow for string storage and in version 1.5.0 introduced the high level ``pd.ArrowDtype`` wrapper.
+
+While these new type systems have brought about many great features, they have surfaced three major problems.
+
+### Problem 1: Inconsistent Type Naming / Behavior
+
+There is no better example of our current type system being problematic than strings. Let's assess the number of string iterations a user could create (this is a non-exhaustive list):
+
+```python
+dtype=object
+dtype=str
+dtype="string"
+dtype=pd.StringDtype()
+dtype=pd.StringDtype("python", na_marker=np.nan)
+dtype=pd.StringDtype("python", na_marker=pd.NA)
```

Comment:
some places you've switched from na_marker to na_value.  should they all be switched?

Reply:
Fixed - thanks for catching